linkerd/admin/src/main/scala/com/twitter/finagle/client/buoyant/ClientStateHandler.scala

```diff
@@ -0,0 +1,41 @@
+package com.twitter.finagle.client.buoyant
+
+import com.twitter.finagle.{Addr, Address, Service}
+import com.twitter.finagle.client.ClientRegistry
+import com.twitter.finagle.http.{MediaType, Request, Response}
+import com.twitter.finagle.loadbalancer.LoadBalancerFactory
+import com.twitter.util.Future
+import io.buoyant.config.Parser
+
+class ClientStateHandler extends Service[Request, Response] {
+
+  private[this] val mapper = Parser.jsonObjectMapper(Nil)
+
+  override def apply(request: Request): Future[Response] = {
+    val entries = ClientRegistry.registrants.map { entry =>
+      val LoadBalancerFactory.Dest(va) = entry.params[LoadBalancerFactory.Dest]
+      va.changes.toFuture().map(entry.addr -> _)
```

Comment:
can we limit this to entries where _entry.addr_ is a `Name` or `Path` so that we don't encode weird things for clients that don't do name resolution?

Reply:
Are we confident that the Var LoadBalancerFactory.Dest is reference counted appropriately?

Might we need to wrap it in `Var.async` when it's constructed?